templates/terraform/constants/cloudbuild_trigger.erb

```diff
@@ -0,0 +1,49 @@
+var (
+	cloudBuildTimeoutRegexp = regexp.MustCompile("\\d+s$")
+)
+
+func stepTimeoutCustomizeDiff(diff *schema.ResourceDiff, v interface{}) error {
+	buildList := diff.Get("build").([]interface{})
+	if len(buildList) == 0 || buildList[0] == nil {
+		return nil
+	}
+	build := buildList[0].(map[string]interface{})
+	buildTimeoutString := build["timeout"].(string)
+
+	matched := cloudBuildTimeoutRegexp.MatchString(buildTimeoutString)
+	if !matched {
+		return fmt.Errorf("Cloud build timeout is not in duration format: %s", buildTimeoutString)
+	}
+
+	buildTimeout, err := strconv.Atoi(buildTimeoutString[0 : len(buildTimeoutString)-1])
```

Comment:
You can potentially avoid this by using time.ParseDuration - this does add the validation of "must be \d+s` ", but we could add that as validation on the field and then have this use time.ParseDuration